crypto.go

```diff
@@ -139,27 +155,58 @@ func verifyCertificateVerify(handshakeBodies []byte, hashAlgorithm HashAlgorithm
 	return errKeySignatureVerifyUnimplemented
 }
 
-func verifyClientCert(cert *x509.Certificate, roots *x509.CertPool) error {
+func loadCerts(rawCertificates [][]byte) ([]*x509.Certificate, error) {
+	certs := make([]*x509.Certificate, 0, len(rawCertificates))
```

Comment:
Minor issue, but you might want to move this below the size check to avoid the unnecessary allocation

Reply:
Changes look good besides this one minor issue.  I'll approve afterwards

Reply:
Fixed!